### PR TITLE
Add support for database schema; Add check for connection.connected to prevent mssql errors

### PR DESCRIPTION
**Pull request by JohnMBarry**

### CODE REVIEW

File: src/store.coffee [hunk 1]
```diff
@@ -16,7 +16,8 @@ module.exports = (session) ->
 		###
 		
 		constructor: (config, options) ->
-			@table = options.table if options?.table
+			@table = "[" + options.table + "]" if options?.table
+			@table = "[" + config.schema + "].#{@table}" if config?.schema
```
Comment (patriksimek) [inline src/store.coffee]: I don't think `config` is a good collection to specify `schema` in. Why not to let user specify schema inside `table` variable together with the table name?

File: src/store.coffee [hunk 2]
```diff
@@ -16,7 +16,8 @@ module.exports = (session) ->
 		###
 		
 		constructor: (config, options) ->
-			@table = options.table if options?.table
+			@table = "[" + options.table + "]" if options?.table
```
Comment (patriksimek) [inline src/store.coffee]: `options.table = "[sessions]"` will break the code.

File: src/store.coffee (hunk 1 above)
Comment (JohnMBarry) [inline src/store.coffee]: Yeah that works.  Would need to add square brackets  or require the user to ensure square brackets are applied if the schema or table has spaces / special characters.

File: src/store.coffee (hunk 2 above)
Comment (JohnMBarry) [inline src/store.coffee]: Ok. So you'd leave it up to the user to use square brackets if the schema or table has spaces / special characters.

**Comment by patriksimek** [comment]
Thanks, releases in 1.5.0.

**Pull request merged**